packages/desktop-client/src/components/reports/spreadsheets/default-spreadsheet.tsx

```diff
@@ -167,10 +171,32 @@ export default function createSpreadsheet({
       const calc = recalculate({ item, months, assets, debts, groupByLabel });
       return { ...calc };
     });
+    const calcDataFiltered = calcData.filter(i =>
+      !showEmpty ? i[balanceTypeOp] !== 0 : true,
+    );
+
+    const colorScale = getColorScale('qualitative');
+    const chooseData = ['Month', 'Year'].includes(groupBy)
+      ? monthData
+      : calcDataFiltered;
+    const legend = chooseData.map((c, index) => {
+      return {
+        name: ['Month', 'Year'].includes(groupBy) ? c.date : c.name,
+        color:
+          graphType === 'DonutGraph'
+            ? colorScale[index % colorScale.length]
+            : ['Month', 'Year'].includes(groupBy)
+            ? balanceTypeOp === 'totalDebts'
+              ? theme.reportsRed
+              : theme.reportsBlue
+            : colorScale[index % colorScale.length],
+      };
+    });
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏ideally we would extract this logic into a separate utility function or hook. Just to keep the code cleaner.

Reply:
I like the idea of having it separate because I'm sure we'll need to adjust it at some point in the future.